File: packages/setpieces-r/setpieces/tests/testthat/test-penalties.R
test_that("penalty_summary counts awarded/scored/missed", {
  m <- sample_match()
  events <- extract_set_pieces(m)
  summary <- penalty_summary(events)

  expect_equal(summary$awarded, 2)
  expect_equal(summary$scored, 1)
  expect_equal(summary$missed, 1)
  expect_equal(summary$saved, 0)
  expect_equal(summary$post, 0)
  expect_equal(summary$conversion_rate, 50.0)
})

test_that("penalty_summary handles no penalties", {
  summary <- penalty_summary(list())
  expect_equal(summary$awarded, 0)
  expect_true(is.na(summary$conversion_rate))
})

File: packages/setpieces-py/tests/test_penalties.py
import setpieces as sp


def test_penalty_summary(match):
    events = sp.extract_set_pieces(match)
    summary = sp.penalty_summary(events)
    assert summary["awarded"] == 2
    assert summary["scored"] == 1
    assert summary["missed"] == 1
    assert summary["saved"] == 0
    assert summary["post"] == 0
    assert summary["conversion_rate"] == 50.0


def test_penalty_summary_empty():
    summary = sp.penalty_summary([])
    assert summary["awarded"] == 0
    assert summary["conversion_rate"] is None
